### PR TITLE
Equal aspect for hic matrix

### PR DESCRIPTION
**Pull request by Phlya**
I think equal aspect looks much better for the Hi-C matrix, makes pixels actually square... Perhaps it can be an argument passed to the track, but IMO equal aspect is a better default. What do you think?

### CODE REVIEW

**Comment by Nanguage** [comment]
Hello, I also noticed this issue before. Unfortunately, I didn't do any Hi-C related work later on, which resulted in a lack of time to maintain this project at the moment. Thank you very much for your contribution!

**Comment by Phlya** [comment]
I think it works fine for me, the only issue (which is simply the limitations of laws of mathematics) is that with a small height of the track the heatmap might not fill the whole width of the frame...

**Comment by Nanguage** [comment]
Can you provide some graphical examples?

**Comment by StefanoCretti** [comment]
Stumbled upon this issue while trying to implement custom matrix tracks for a project I am working on. I think @Phlya is referring to the results which can be obtained from the following snippet of code.

```
import coolbox.api as ca  # 0.3.8 patched according to @Phyla's suggestion

# File is a subset of the 10kb cooler generated starting from the pairs at
# https://data.4dnucleome.org/files-processed/4DNFIIG4IWKW/
# Can provide if needed, but any region of any file should be the same
COOL_PATH: str = "test_files/chr17_71000000_74000000.cool"
REGION: str = "chr17:71500000-73500000"

frame = ca.ChromName() + ca.Cool(COOL_PATH, height=5) + ca.InsuScore(ca.Cool(COOL_PATH))
fig = frame.plot(REGION)
fig.savefig("test_5.png")
```
This will yield the following image:

test_5

When the height of the track is too small, the image will not touch the expected borders. This is a major issue when displaying multiple tracks, since they will not be aligned.

On the other hand, by running the same code but with `height=20`, the result is as follows:

test_20

In this case, since there is enough height, the left and right border will be touched, though there can be some annoying white space below the plot. I could not find an easy way to remove the space and it makes difficult to plot multiple tracks in a compact manner.

My only idea to solve the issue would be to force the track height to be, by default, some function of track width and `depth_ratio`, which would result in neither of the white spaces. I tried tinkering a bit with it but so far I was not able to come up with a solution, due to the underlying rotated matrix complicating things as well as the interval being internally extended.

**Comment by StefanoCretti** [comment]
Btw, I found a way to fill the space correctly when the aspect ratio is set to "equal".
TLDR:

- In `Frame.get_track_height` the `frame_width` variable does not take into account the margins that are set by `subplot_adjust` in `Frame.plot`. Should be rescaled ads follows.
```python
margins = self.properties["margins"]
margins_fraction = margins["right"] - margins["left"]
frame_width = (
    self.properties["width"]
    * self.properties["width_ratios"][1]
    * margins_fraction
)
```
- When setting up the `GridSpec` in `Frame.plot`, `wspace=0.01` should be omitted

With these two changes we have the expected behavior.
test


This works well with `depth_ratio` but obviously still breaks if the height of the track is forced manually.
@Nanguage if you are interested in a PR let me know.

**Comment by Nanguage** [comment]
Hi @StefanoCretti, Thank you for your effort! I am very interested in PR. Perhaps a parameter (such as equal_aspect=True) can be added to control whether to use this code, in order to avoid conflicts with directly setting the height.

**Comment by StefanoCretti** [comment]
Proposed change was addressed in PR https://github.com/GangCaoLab/CoolBox/pull/104. @Nanguage you can probably close this thread.
